common/osqp_interface/test/test_osqp_interface.cpp

```diff
@@ -27,49 +27,49 @@ using autoware::common::osqp::float64_t;
 // cppcheck-suppress syntaxError
 TEST(TestOsqpInterface, BasicQp) {
   auto check_result =
-    [](const std::tuple<std::vector<float64_t>, std::vector<float64_t>, int, int> & result) {
+    [](const std::tuple<std::vector<float64_t>, std::vector<float64_t>, int, int, int> & result) {
       EXPECT_EQ(std::get<2>(result), 1);
       EXPECT_EQ(std::get<3>(result), 1);
       ASSERT_EQ(std::get<0>(result).size(), size_t(2));
       ASSERT_EQ(std::get<1>(result).size(), size_t(2));
       EXPECT_DOUBLE_EQ(std::get<0>(result)[0], 0.3);
       EXPECT_DOUBLE_EQ(std::get<0>(result)[1], 0.7);
       EXPECT_DOUBLE_EQ(std::get<1>(result)[0], -2.9);
-      EXPECT_NEAR(std::get<1>(result)[1], 0.2, 1e-6);
+      EXPECT_NEAR(std::get<1>(result)[1], 0.0, 1e-6);
```

Comment:
Without this change, the test failed, which I'm not sure why now.

Reply:
@rej55 Do you have any idea about this change (from 0.2 to 0.0)? For the code, I just changed from `Eigen::MatrixXd A(2, 4);` to `Eigen::MatrixXd A(4, 2);` as following diffs on this page.

Reply:
@takayuki5168 
Since the problem has changed, I think it is proper to change the solution.
However, I wonder why the previous code was working.... :thinking: 

In this problem, the size of Lagrange multiplier should be 4 because the row size of the constraint matrix A is 4.
However, in this test code, it expects the size of Lagrange multiplier as 2.
Could you tell me the size of the resultant Lagrange multiplier?
https://github.com/autowarefoundation/autoware.universe/blob/f108d12facabdd01c448094814a058a8ac35ba49/common/osqp_interface/test/test_osqp_interface.cpp#L34

Reply:
@rej55 
The size of lagrange multiplier does not change, and it is 2. Not 4 as you pointed :thinking: 
https://github.com/autowarefoundation/autoware.universe/blob/f108d12facabdd01c448094814a058a8ac35ba49/common/osqp_interface/test/test_osqp_interface.cpp#L34

Reply:
Seems this problem occurs like below:

 - CSC matrix does not check the dimension consistency.
 - Then, the A(2, 4) matrix was considered as A(2, 2) and the half of the A matrix was ignored in the optimization.
 - But the optimal solution of the prime problem was accidentally unchanged. This is because the optimal solution does not on the ignored constraints.
 - The dual variable had a different dimension from the original solution, but this was not checked in the test.

So this will be solved by adding the full evaluation of the dual variables.

Reply:
I push some fix and changes here.
https://github.com/tier4/autoware.universe/tree/feature/osqp-add-functions